tests/test_gcloud.py

```diff
@@ -0,0 +1,210 @@
+
+        self.assertEqual(len(dirs), 2)
+        for directory in ["some", "other"]:
+            self.assertTrue(directory in dirs,
+                            """ "%s" not in directory list "%s".""" % (
+                                directory, dirs))
+
+        self.assertEqual(len(files), 2)
+        for filename in ["2.txt", "4.txt"]:
+            self.assertTrue(filename in files,
+                            """ "%s" not in file list "%s".""" % (
+                                filename, files))
+
+    def test_listdir_subdir(self):
+        file_names = ["some/path/1.txt", "some/2.txt"]
+
+        self.storage._bucket = mock.MagicMock()
+        self.storage._bucket.list_blobs.return_value = []
+        for name in file_names:
+            blob = mock.MagicMock(spec=Blob)
+            blob.name = name
+            self.storage._bucket.list_blobs.return_value.append(blob)
+
+        dirs, files = self.storage.listdir('some/')
+
+        self.assertEqual(len(dirs), 1)
+        self.assertTrue('path' in dirs,
+                        """ "path" not in directory list "%s".""" % (dirs,))
+
+        self.assertEqual(len(files), 1)
+        self.assertTrue('2.txt' in files,
+                        """ "2.txt" not in files list "%s".""" % (files,))
+
+    def test_size(self):
+        size = 1234
+
+        self.storage._bucket = mock.MagicMock()
+        blob = mock.MagicMock()
+        blob.size = size
+        self.storage._bucket.get_blob.return_value = blob
+
+        self.assertEqual(self.storage.size(self.filename), size)
+        self.storage._bucket.get_blob.assert_called_with(self.filename)
+
+    def test_modified_time(self):
+        date = datetime.datetime(2017, 1, 2, 3, 4, 5, 678)
+
+        self.storage._bucket = mock.MagicMock()
+        blob = mock.MagicMock()
+        blob.updated = date
+        self.storage._bucket.get_blob.return_value = blob
+
+        self.assertEqual(self.storage.modified_time(self.filename), date)
+        self.storage._bucket.get_blob.assert_called_with(self.filename)
+
+    def test_url(self):
+        url = 'https://example.com/mah-bukkit/{}'.format(self.filename)
+
+        self.storage._bucket = mock.MagicMock()
+        blob = mock.MagicMock()
+        blob.public_url = url
+        self.storage._bucket.get_blob.return_value = blob
+
+        self.assertEqual(self.storage.url(self.filename), url)
+        self.storage._bucket.get_blob.assert_called_with(self.filename)
+
+    def test_get_available_name(self):
+        self.storage.file_overwrite = True
+        self.assertEqual(self.storage.get_available_name(self.filename), self.filename)
+
+        self.storage._bucket = mock.MagicMock()
+        self.storage._bucket.get_blob.return_value = None
+        self.storage.file_overwrite = False
+        self.assertEqual(self.storage.get_available_name(self.filename), self.filename)
+        self.storage._bucket.get_blob.assert_called_with(self.filename)
```

Comment:
Let's add a unicode test name.  Would like to not have Python 2.7 issues aka [this bug](https://github.com/jschneier/django-storages/pull/217).